Typing: creating Teleport with h and RawSlots throws

### Version
3.0.2

### Reproduction link
[https://codesandbox.io/s/magical-brook-6vtfm?file=/src/components/HelloWorld.vue](https://codesandbox.io/s/magical-brook-6vtfm?file=/src/components/HelloWorld.vue)

### Steps to reproduce
Create a Teleport VNode using h with an object of slots as third parameter, for example in a setup function:

```ts
setup() {
  return () => {
    h(
      Teleport,
      { to: "#portal" },
      {
        default() {
          return 'Child';
        },
      }
    );
  };
},
```

### What is expected?
The code shouldn’t throw typing errors. At least that’s what I’m thinking since the code works, but I might be missing some edge-case problems this syntax could cause?

### What is actually happening?
The following typing error is thrown by TypeScript:

```
semantic error TS2769: No overload matches this call.
  The last overload gave the following error.
    Argument of type '{ new (): { $props: VNodeProps & TeleportProps; }; __isTeleport: true; }' is not assignable to parameter of type 'DefineComponent<{ to: string; }, {}, {}, Record<string, ComputedGetter<any> | WritableComputedOptions<any>>, MethodOptions, ... 6 more ..., {}>'.
      Type '{ new (): { $props: VNodeProps & TeleportProps; }; __isTeleport: true; }' is not assignable to type 'ComponentPublicInstanceConstructor<{ $: ComponentInternalInstance; $data: {}; $props: Partial<{}> & Pick<Readonly<{} & { to?: string | undefined; }> & VNodeProps & AllowedComponentProps & ComponentCustomProps, "key" | ... 9 more ... | "to">; ... 10 more ...; $watch(source: string | Function, cb: Function, options?: ...'.
        Types of property '__isTeleport' are incompatible.
          Type 'true' is not assignable to type 'undefined'.
```

<!-- generated by vue-issues. DO NOT REMOVE -->

--- a/packages/runtime-core/src/h.ts
+++ b/packages/runtime-core/src/h.ts
@@ -96,7 +96,7 @@ export function h(
 export function h(
   type: typeof Teleport,
   props: RawProps & TeleportProps,
-  children: RawChildren
+  children: RawChildren | RawSlots
 ): VNode
 
 // suspense

--- a/test-dts/h.test-d.ts
+++ b/test-dts/h.test-d.ts
@@ -47,6 +47,7 @@ describe('h inference w/ Fragment', () => {
 
 describe('h inference w/ Teleport', () => {
   h(Teleport, { to: '#foo' }, 'hello')
+  h(Teleport, { to: '#foo' }, { default() {} })
   // @ts-expect-error
   expectError(h(Teleport))
   // @ts-expect-error

